### PR TITLE
Initial version of jsbuilder tool

### PR DESCRIPTION
**Pull request by jimmyaxod**
This introduces a jsbuilder tool which creates a .wasm file from js.

The rust portion simply passes on data to js, where it should *then* be decoded/encoded as per the signature encoding.

The javascript should have the following exported.
runFunction(data number[]): number[]

(global as any).Exports = {
  main: mainFunction,
  run: runFunction,
}

The rust portion also exports a function to call back to next(), which again accepts bytes and returns bytes.

newData = (global as any).scale_fn_next(data)

### CODE REVIEW

**Comment by jimmyaxod** [comment]
Have removed buffers from the rust portion. These are now in js.
New function exposed to js - getaddr which returns the address of an ArrayBuffer.
The Resize function now routes through to js.
Next now takes [ptr, len] and returns packed.

File: ts/compile/builder/crates/core/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,269 @@
+#![allow(dead_code)]
+#![allow(unused_imports)]
```
Comment (ShivanshVij) [inline ts/compile/builder/crates/core/src/main.rs:2]: Same as above

**Review by ShivanshVij — APPROVED** [review]
Minor changes plus what we talked about during our 1:1

File: ts/compile/builder/crates/core/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+};
+use std::os::raw::{c_char, c_int, c_void};
+
+use std::io::{self, Cursor, Read, Write};
+use std::ffi::CString;
+
+extern crate wee_alloc;
+
+#[cfg(not(test))]
+#[global_allocator]
```
[the remark this answers is not in the document]
Comment (jimmyaxod) [reply ts/compile/builder/crates/core/src/main.rs]: 5ada341efa2e865f4dad6f2ca5e488a37158956a

**Pull request closed without merging**